src/wp-includes/class-wp-block.php

```diff
@@ -265,6 +265,15 @@ private function process_block_bindings() {
 				continue;
 			}
 
+			// Add the necessary context defined by the source.
+			if ( ! empty( $block_binding_source->uses_context ) ) {
```

Comment:
It didn't occur to me that we could inject the context just as we are about to compute it. Great idea!

I think the fact that we have a formal block binding source instance helps a ton here.